url3ed4973106_kosmasK_RecordPlayer_tgz-pJ8-es_uc3m_recordplayer_test_TestJ8/src/es/uc3m/eda/tree/BSTree.java

```diff
@@ -0,0 +1,184 @@
+package es.uc3m.eda.tree;
+
+import es.uc3m.eda.list.IList;
+import es.uc3m.eda.list.singlelink.SList;
+import es.uc3m.eda.tree.view.BSTreeView;
+
+public class BSTree<K extends Comparable<K>, E> implements IBSTree<K, E> {
+	BSTNode<K, E> root;
+
+	public BSTree() {
+		this.root = null;
+	}
+
+	public BSTree(BSTNode<K, E> node) {
+		this.root = node;
+	}
+
+	@Override
+	public BSTNode<K, E> getRoot() {
+		return this.root;
+	}
+
+	@Override
+	public void insert(K key, E elem) {
+		BSTNode<K, E> newNode = new BSTNode<K, E>(key, elem);
+		if (this.root == null) {
+			this.root = newNode;
+			return;
+		}
+		insert(newNode, this.root);
+	}
+
+	private void insert(BSTNode<K, E> newNode, BSTNode<K, E> node) {
+		K key = newNode.key;
+		// if (key.compareTo(node.key) == 0) {
+		// System.out
+		// .println("Cannot insert, the key already exists");
+		// return;
+		// }
+		if (key.compareTo(node.key) < 0) {
+			if (node.leftChild == null) {
+				node.leftChild = newNode;
+				newNode.parent = node;
+				return;
+			} else {
+				insert(newNode, node.leftChild);
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 41:47, 48:56.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.